staging/src/k8s.io/client-go/transport/token_source_test.go

```diff
@@ -141,15 +141,29 @@ func TestCachingTokenSourceRace(t *testing.T) {
 		var wg sync.WaitGroup
 		wg.Add(100)
 
+		errs := make(chan error, 100)
 		for i := 0; i < 100; i++ {
 			go func() {
 				defer wg.Done()
-				if _, err := ts.Token(); err != nil {
-					t.Fatalf("err: %v", err)
+
+				_, err := ts.Token()
+				if err != nil {
+					errs <- err
 				}
 			}()
 		}
-		wg.Wait()
+
+		go func() { // wait in another go-routine to close channel
```

Comment:
if errs is closed here, the line 161: for err := range errs {} may read from the closed channel, right?

Reply:
yes

Reply:
reading from closed channel would return nothing, how do you expect to read err out of closed errs channel at line: 161?

Reply:
the line 161:for err := range errs {}  execute  before the errs closed.

Reply:
in addition, after a channel closed, we also can read the obj in the channel.

Reply:
i think there is still race here. could you run the test 1000 times to see if there is any failure?

bazel test --runs_per_test=1000 //staging/src/k8s.io/client-go/...

Reply:
command "bazel test --runs_per_test=1000 //staging/src/k8s.io/client-go/transport/..." show:

```
INFO: Analyzed 7 targets (0 packages loaded, 0 targets configured).
INFO: Found 6 targets and 1 test target...
INFO: Elapsed time: 3.664s, Critical Path: 0.55s
INFO: 1000 processes: 1000 linux-sandbox.
INFO: Build completed successfully, 1001 total actions
//staging/src/k8s.io/client-go/transport:go_default_test                 PASSED in 0.3s
  Stats over 1000 runs: max = 0.3s, min = 0.1s, avg = 0.2s, dev = 0.0s

Executed 1 out of 1 test: 1 test passes.
INFO: Build completed successfully, 1001 total actions
```